python/ray/_private/client_mode_hook.py

```diff
@@ -69,6 +69,30 @@ def client_mode_should_convert():
     return client_mode_enabled and _client_hook_enabled
 
 
+def client_mode_wrap(func):
+    """Wraps a function called during client mode for execution as a remote
+    task.
+
+    Can be used to implement public features of ray client which do not
+    belong in the main ray API (`ray.*`), yet require server-side execution.
+    An example is the creation of placement groups:
+    `ray.util.placement_group.placement_group()`. When called on the client
+    side, this function is wrapped in a task to facilitate interaction with
+    the GCS.
+    """
+    from ray.util.client import ray
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if client_mode_should_convert():
+            f = ray.remote(num_cpus=0)(func)
```

Comment:
I'm still concerned about why we need to submit these tasks to the cluster, it just seems kinda unnecessary and pretty decent chunk of overhead/complication. I understand that we can't run these functions on the client, but the client server is a driver in the cluster so it should be fine right?

Reply:
The goal of this PR is to make public placement group APIs available on the client.

Reply:
So a user should be able to do the following locally
```
import ray
from ray.util.placement_group import placement_group
ray.util.connect(...)
pg = placement_group(...)
...
```
Currently, that would throw an error [here](https://github.com/ray-project/ray/blob/fe06642df0e4b88ac315028ba7de2855cd27a710/python/ray/util/placement_group.py#L177)

Reply:
@wuisawesome I agree that this is slower than having a  `gRPC` message for each of these functions, but I don't think that the minor performance difference is worth the additional complexity added to the client + server (namely having to create and manage `ClientPlacementGroups`).

Reply:
@wuisawesome Is this ok to resolve for now?

Reply:
Does this seem like an abuse of Client tasks? Is this iffiness of this that tasks should be used for heavy computations 
(rather than for simple RPCs which is pretty much what's going on here) ?

Reply:
I guess there's a reason `num_cpus=1` is the default for a task, rather than `num_cpus=0`? What's the correct use case (if any) for a `num_cpus=0` task?

Reply:
> tasks should be used for heavy computations

There is definitely more overhead with submitting a Ray task (as opposed to a RPC call). That being said I think the extra overhead isn't that big of a deal given that if people are trying to create many, many placement groups, the real bottleneck will all likelihood be in waiting for the cluster to scale to the appropriate size to support all the placement groups. 

> What's the correct use case (if any) for a num_cpus=0 task?

For computationally light tasks. The functions being wrapped are either `getter/setter` methods (tiny overhead) or are waiting for PG creation (which can be thought of as a 'node-startup'-bound operation).